### PR TITLE
Add test PR job

### PR DESCRIPTION
**Pull request by fejta**
Testing out different ways of handling https://github.com/kubernetes/test-infra/pull/439

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/459)

### CODE REVIEW

File: jenkins/checkout.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import subprocess
+
+
+def Checkout(git, org, repo, branch, pull):
+    if bool(branch) ^ bool(pull) == False:
```
Comment (spxtr) [inline jenkins/checkout.py]: `if bool(branch) == bool(pull):` seems nicer to me, and I would update the error to say "Must specify exactly one of ...".

Comment (fejta) [inline jenkins/checkout.py]: Done

**Comment by spxtr** [comment]
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[jenkins/checkout.py, line 26 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMM4_r91qWuO7ITYFJ:-KQMM4_r91qWuO7ITYFK:b-6d36bf) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/checkout.py#L26)):*
> ```Python
>         raise ValueError('Must specify one of --branch or --pull')
>     if pull:
>         ref = ['+refs/pull/%d/merge' % pull]
> ```

Why a list?

---

*[jenkins/pr-e2e-gce.sh, line 16 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMN4wMcJvPuHndbdyi:-KQMN4wMcJvPuHndbdyj:b-76v34x) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/pr-e2e-gce.sh#L16)):*
> ```Shell
> # limitations under the License.
> 
> set -ex
> ```

For consistency I'd do `errexit`, `nounset`, and `pipefail` on their own lines.

---

*[jenkins/pr-e2e-gce.sh, line 17 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMNBZCae1vtUOZkYio:-KQMNBZCae1vtUOZkYip:bf0e7n) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/pr-e2e-gce.sh#L17)):*
> ```Shell
> 
> set -ex
> checkout="${PWD}/test-infra/jenkins/checkout.py"
> ```

Do `checkout` and `runner` actually need to be variables? I think you only use them once. Also, I think we prefer declaring them readonly here.

---

*[jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml, line 34 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMO-7GAXPQjzddzqoa:-KQMO-7GAXPQjzddzqob:b-124kdj) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml#L34)):*
> ```YAML
>             status-url: 'https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/${{ghprbPullId}}/${{JOB_NAME}}/${{BUILD_NUMBER}}/'
>             status-add-test-results: true
>             # This should roughly line up with kubernetes-maintainers.
> ```

Comment is incorrect.

---

*[jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml, line 94 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMNxxPmpEo4uGHssq_:-KQMNxxPmpEo4uGHssqa:b-2co5do) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml#L94)):*
> ```YAML
>     repo-name: 'kubernetes/kubernetes'
>     suffix:
>       - test-unit-integration: # kubernetes-pull-test-unit-integration
> ```

Comments aren't right.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459)*

**Comment by fejta** [comment]
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[jenkins/checkout.py, line 26 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMM4_r91qWuO7ITYFJ:-KQNMtHPNegU6AXhizQZ:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/checkout.py#L26)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> Why a list?
</details>
Done.

---

*[jenkins/pr-e2e-gce.sh, line 16 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMN4wMcJvPuHndbdyi:-KQNMtzG1FDLfCdyk_MF:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/pr-e2e-gce.sh#L16)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> For consistency I'd do `errexit`, `nounset`, and `pipefail` on their own lines.
</details>
Done.

---

*[jenkins/pr-e2e-gce.sh, line 17 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMNBZCae1vtUOZkYio:-KQNMuhvnU5eycK7EE1T:b-mb59am) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/pr-e2e-gce.sh#L17)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> Do `checkout` and `runner` actually need to be variables? I think you only use them once. Also, I think we prefer declaring them readonly here.
</details>
Yes, this is easier given that we `cd`

---

*[jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml, line 34 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMO-7GAXPQjzddzqoa:-KQNN3M-wUstde4ONK7k:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml#L34)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> Comment is incorrect.
</details>
Done.

---

*[jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml, line 94 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMNxxPmpEo4uGHssq_:-KQNN3msYVV6SW9z23-M:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml#L94)):*
<details><summary><i>Previously, spxtr (Joe Finney) wrote…</i></summary>
> Comments aren't right.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459)*

**Comment by spxtr** [comment]
This seems reasonable. It's a bother that even a test job like this will add yet another "ok to test" message...

---

Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[jenkins/pr-e2e-gce.sh, line 17 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-KQMNBZCae1vtUOZkYio:-KQNOz-4rBOMQ45rJ4ID:b7d0x1k) ([raw file](https://github.com/kubernetes/test-infra/blob/a1e829bb4521c5c9271025cf0928d0234e4f3621/jenkins/pr-e2e-gce.sh#L17)):*
<details><summary><i>Previously, fejta (Erick Fejta) wrote…</i></summary>
> Yes, this is easier given that we `cd`
</details>
Oh I see. Ideally our scripts can be run from anywhere, but that's probably too much to dream for.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/459#-:-KQNQDp6_HAkePDDer7q:b4a7oew)*

**Comment by fejta** [comment]
Adding LGTM and merging per Joe's comments.

**Comment by ixdy** [comment]
We may want to refactor the jenkins/ directory before we add even more scripts there. i.e. the maintenance/utility scripts should probably be separate from the test runners.

**Comment by ixdy** [comment]
also, I think this just made Jenkins spam every open PR.

File: jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        - github-pull-request:
+            # This is the Jenkins GHPRB plugin ID, not the actual github token.
+            auth-id: 'f8e31bc1-9abb-460a-a2ca-9c4aae3ca4e8'
+            only-trigger-phrase: true
+            trigger-phrase: '(?is).*@fejta do a little dance.*'
+            cron: 'H/2 * * * *'
+            status-context: Jenkins {status-context}
+            status-url: 'https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/${{ghprbPullId}}/${{JOB_NAME}}/${{BUILD_NUMBER}}/'
+            status-add-test-results: true
+            admin-list:
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml]: can we at least add the kubernetes org as a whitelist? this is spamming *every* new PR twice.

Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins-pull/fejta-pull.yaml]: Sure makes a lot of sense to me

**Pull request merged**